nailgun/entities.py

```diff
@@ -2879,6 +2879,49 @@ class Meta(object):
         api_path = 'api/v2/smart_proxies'
         server_modes = ('sat')
 
+    def path(self, which=None):
+        """Extend ``nailgun.entity_mixins.Entity.path``.
+
+        The format of the returned path depends on the value of ``which``:
+
+        refresh
+            /katello/api/v2/smart_proxies/:id/refresh
+
+        """
+        if which in ('refresh',):
+            return '{0}/{1}'.format(
+                super(SmartProxy, self).path(which='self'),
+                which
+            )
+        return super(SmartProxy, self).path(which)
+
+    def refresh(self, synchronous=True):
+        """Refresh Capsule features
+
+        .. NOTE:: The ``synchronous`` argument has no effect in certain
+            versions of Satellite. See `Bugzilla #1199150
+            <https://bugzilla.redhat.com/show_bug.cgi?id=1199150>`_.
+
+        :param synchronous: What should happen if the server returns an HTTP
+            202 (accepted) status code? Wait for the task to complete if
+            ``True``. Immediately return the server's reponse otherwise.
+        :returns: The server's JSON-decoded response.
+
+        """
+        response = client.put(
+            self.path('refresh'),
+            {},
+            auth=self._server_config.auth,
+            verify=self._server_config.verify,
+        )
+        response.raise_for_status()
+        if synchronous is True and response.status_code == ACCEPTED:
+            return ForemanTask(
+                self._server_config,
+                id=response.json()['id'],
+            ).poll()
+        return response.json()
```

Comment:
This response handling code is great for now. As soon as #50 is merged, you'll be able to just do this intead:

```python
response = client.put(…)
return _handle_response(response, self._server_config, synchronous)
```